### PR TITLE
`getOwnPropertyDescriptor`: not actually supported in IE 8

### LINKED ISSUE #5202: `Object.getOwnPropertyDescriptor` does not have full support in IE 8
Specifically, `Object.getOwnPropertyDescriptor(arguments, '')` will throw, where it should not.

I found the line that needs updating: https://github.com/mdn/browser-compat-data/blob/7f8137d2b78bc1ec37f8eb11e27c128772e6c09e/javascript/builtins/Object.json#L768 but i'm not sure what to put there, since it's not not *added* in IE 8, it's not experimental, or deprecated - it's just got partial support.

### PR DESCRIPTION
**Pull request by ljharb**
follows the precedent set by defineProperty.

Fixes #5202.

A checklist to help your pull request get merged faster:
- [x] Summarize your changes
- [ ] Data: link to resources that verify support information (such as browser's docs, changelogs, source control, bug trackers, and tests)
- [ ] Data: if you tested something, describe how you tested with details like browser and version
- [ ] Review the results of the linter and fix problems reported (If you need help, please ask in a comment!)
- [ ] Link to related issues or pull requests, if any

### CODE REVIEW

File: javascript/builtins/Object.json [hunk 1]
```diff
@@ -819,7 +819,8 @@
                 "version_added": "4"
               },
               "ie": {
-                "version_added": "8"
+                "version_added": "9",
```
Comment (vinyldarkscratch) [inline javascript/builtins/Object.json]: There **was** actually support in IE 8, it just wasn't great, so let's undo the change to this line.

```suggestion
                "version_added": "8",
```

File: javascript/builtins/Object.json [hunk 2]
```diff
@@ -819,7 +819,8 @@
...
+                "notes": "Also supported in Internet Explorer 8, but only on DOM objects and with some non-standard behaviors."
```
Comment (vinyldarkscratch) [inline javascript/builtins/Object.json]: I think it would help to explain **what** the non-standard behaviors were, rather than just saying there were some.  The more details, the better!

```suggestion
                "notes": "Support in Internet Explorer 8 was only present on DOM objects with some non-standard behaviors [Add Explanation].  This behavior was fixed in Internet Explorer 9."
```

File: javascript/builtins/Object.json (hunk 1 above)
Comment (ljharb) [reply javascript/builtins/Object.json]: Should i change defineProperty then, too?

File: javascript/builtins/Object.json (hunk 2 above)
Comment (ljharb) [reply javascript/builtins/Object.json]: I have no idea what they are. I copied this phrasing from `defineProperty`.

**Comment by vinyldarkscratch** [comment]
I'll be re-reviewing this PR tomorrow or later this week, just gotta finish packing for All Hands 2020 tonight.  Just wanted to let you know, though: you don't need to force push your PRs into a single commit (if that's why you had force pushed), as we strictly squash merge for this repository.  :wink:

**Comment by ljharb** [comment]
I need to because I care about proper git hygiene :-) the final merge method doesn't relieve me of that moral obligation.

Thanks, look forward to your review!

File: javascript/builtins/Object.json [hunk 3]
```diff
@@ -440,8 +440,9 @@
                 "version_added": "4"
               },
               "ie": {
-                "version_added": "9",
-                "notes": "Also supported in Internet Explorer 8, but only on DOM objects and with some non-standard behaviors."
+                "version_added": "8",
+                "partial_implementation": true,
+                "notes": "Only supported on DOM objects and with some non-standard behaviors."
```
Comment (vinyldarkscratch) [inline javascript/builtins/Object.json]:
```suggestion
                "notes": "In Internet Explorer 8, this was only supported on DOM objects and with some non-standard behaviors.  This was later fixed in Internet Explorer 9."
```

File: javascript/builtins/Object.json (hunk 2 above)
Comment (vinyldarkscratch) [reply javascript/builtins/Object.json]: Hmm...  I guess we’ll have to look into what that behavior is, then, since it’s not too helpful to mention there **is** non-standard behavior and not explain what that behavior is.  I‘ll see if I can dig into it!
